### PR TITLE
LLVM and SPIRV-LLVM-Translator pulldown (WW41 2024)

### PR DESCRIPTION
**Pull request by iclsrc**
LLVM: llvm/llvm-project@2f50b280dc8e995ef67ad31a5f71adc4c270890d
SPIRV-LLVM-Translator: KhronosGroup/SPIRV-LLVM-Translator@d3e72db7e0d74f4

### CODE REVIEW

**Comment by jsji** [comment]
This is ready for review.
- [ ] [Use Basic Blocker Iterator instead](https://github.com/intel/llvm/pull/15669/commits/53562fbe8c248e76b63d3e42184546c9775fb822) @intel/dpcpp-kernel-fusion-reviewers
- [ ] [Update CodeSectionINTEL tests due to EntryPoint customization](https://github.com/intel/llvm/pull/15669/commits/6d84f3a17f1ffb0b2f29029f54cba448f13bf86d) @intel/dpcpp-spirv-reviewers 
- [ ] [[bindless images][Cuda] XFAIL mipmap_read_3D due to type mismatch](https://github.com/intel/llvm/pull/15669/commits/457b79d43112878a4b211af6405ef6accf24f13f) @intel/bindless-images-reviewers

File: llvm-spirv/test/extensions/INTEL/SPV_INTEL_function_pointers/CodeSectionINTEL/alias.ll [hunk 1]
```diff
@@ -10,11 +10,11 @@ target triple = "spir64-unknown-unknown"
 ; when used since they can't be translated directly.
 
 ; CHECK-SPIRV-DAG: Name [[#FOO:]] "foo"
-; CHECK-SPIRV-DAG: Name [[#BAR:]] "bar"
+; CHECK-SPIRV-DAG: EntryPoint [[#]] [[#BAR:]] "bar"
 ; CHECK-SPIRV-DAG: Name [[#Y:]] "y"
 ; CHECK-SPIRV-DAG: Name [[#FOOPTR:]] "foo.alias"
 ; CHECK-SPIRV-DAG: Decorate [[#FOO]] LinkageAttributes "foo" Export
-; CHECK-SPIRV-DAG: Decorate [[#BAR]] LinkageAttributes "bar" Export
+; INTEL-CHECK-SPIRV-DAG: Decorate [[#BAR]] LinkageAttributes "bar" Export
```
Comment (AlexeySachkov) [inline llvm-spirv/test/extensions/INTEL/SPV_INTEL_function_pointers/CodeSectionINTEL/alias.ll]: I don't see any difference in the `CHECK` line, do we really need the prefix?

Comment (jsji) [reply llvm-spirv/test/extensions/INTEL/SPV_INTEL_function_pointers/CodeSectionINTEL/alias.ll]: The prefix is actually to *skip* this line -- no attributes generated.

**Review by AlexeySachkov — APPROVED** [review]

**Comment by jsji** [comment]
@intel/llvm-gatekeepers Please help to issue a /merge.  The dev ci and AMD failures are irrelevant, also failing on other PRs.
https://github.com/intel/llvm/issues/15727 is a follow up for bindless image test.

**Review by bb-sycl — APPROVED** [review]

**Comment by bb-sycl** [comment]
Thu 17 Oct 2024 04:10:01 PM UTC --- Start to merge the commit into sycl branch. It will take several minutes.

**Comment by bb-sycl** [comment]
Thu 17 Oct 2024 04:14:51 PM UTC --- Merge the branch in this PR to base automatically. Will close the PR later.

**Pull request merged**